dvc/cli.py

```diff
@@ -102,11 +123,23 @@ def _find_parser(parser, cmd_cls):
 class DvcParser(argparse.ArgumentParser):
     """Custom parser class for dvc CLI."""
 
+    cmd_choices = []
+
     def error(self, message, cmd_cls=None):  # pylint: disable=arguments-differ
         logger.error(message)
         _find_parser(self, cmd_cls)
 
     def parse_args(self, args=None, namespace=None):
+        # NOTE: this is a custom check to see if any suggestions can
+        # be displayed to users in case of small typos
+        # E.g. `dvc commti` would display
+        # `The most similar command(s) are commit`
+        if args and args[0] not in self.cmd_choices:
+            cmd_suggestions = _find_cmd_suggestions(args[0], self.cmd_choices)
+            if cmd_suggestions:
+                sys.stderr.write(cmd_suggestions)
+                sys.exit(2)
```

Comment:
We generally prefer using the logger rather than writing directly to stderr, and in the same vein, it will be cleaner for us to use the existing exception handling for argument parsing errors, rather than exiting immediately

```suggestion
                logger.error(cmd_suggestions)
                raise DvcParserError
```

Reply:
Sounds good. I noticed there was an `error()` method on the `DvcParser` class but didn't use it because this is where the full usage statement comes from.

Should the full usage statement not be used at all?

Reply:
Right, in this case I don't think we want to include the full usage statement here, so using `logger.error` instead of `self.error` is fine. If the invalid command looks like a small typo, we can just include `See 'dvc --help'` with the new suggested commands error message.